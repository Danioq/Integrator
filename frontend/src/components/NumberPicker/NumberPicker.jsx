import React, {useEffect} from 'react';
import styled from 'styled-components';

let Main = styled.main`
  margin: auto;
  padding: 20px;
`;

export default function NumberPicker(props) {
  const getData = props.getData;
  let actualValue = (event) => {
    document.getElementById('value').innerHTML = event.target.value;
  }
  let handleClick = (event) => {
    event.preventDefault();
    const value = document.getElementById('quantity').value;
    getData(value);
  }
  useEffect(() => {
    document.getElementById('value').innerHTML = document.getElementById('quantity').value;
  }, [])

  return <Main>
    <p id='value'></p>
    <form>
      <label htmlFor="quantity">Wielkość Grupy</label>
      <input type="range" id='quantity' name="quantity" min="2" max="5" onInput={actualValue} />
      <input type="submit" value="Wybierz" onClick={handleClick} />
    </form>
  </Main>
}