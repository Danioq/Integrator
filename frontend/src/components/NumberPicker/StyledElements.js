import styled from 'styled-components';

export let Submit = styled.input.attrs(() =>({
  type: 'submit',
  value: 'Wybierz'
}))`
    display: block;
    margin-left: auto;
    margin-right: auto;
    font-weight: 700;
    font-size: 1.1em;
    background-color: #03e008;
    color: white;
    margin-top: 20px;
    width: 150px;
    height: 40px;
    border-radius: 20px;
    border: none;
    box-shadow: 1px 1px 1px 1px #03c008;
    &:focus{
      box-shadow: none;
      outline: none;
    }
`;

export let Slider = styled.input.attrs(() => ({
  type: 'range',
  id: 'quantity',
  name: 'quantiy', 
  min: '2',
  max: '5'
}))`
    width: 200px;
    height:30px;
    cursor: pointer;
    appearance: none;
    background-color: transparent;
    &:focus {
      outline: none;
    }
    &::-webkit-slider-runnable-track {
      background: white;
      height:30px;
      border-radius: 15px;
      height: 30px
    }
    &::-moz-range-track {
      height: 100%;
      background: white;
      border-radius: 15px;
      height: 30px
    }
    &::-webkit-slider-thumb 
    {
      height: 30px;
      width: 30px;
      border-radius: 15px;
      border: none;
      margin: 0px;

      background-color: #03e008;
      -webkit-appearance: none;
    }
    &::-moz-range-thumb {
      height: 30px;
      width: 30px;
      border-radius: 15px;
      border: none;
      background-color: #03e008;
      -webkit-appearance: none;
    }
`;

export let Label = styled.label.attrs(() => ({
  htmlFor: 'quantity',
}))`
  font-size: 2.5em;
  font-family: Arial, Helvetica, sans-serif;
  font-weight: bold;
  color: #03c008;
`;

export let Value = styled.p.attrs(() => ({
    id: 'value',
}))`
    font-size: 2.5em;
    color: #03c008;
    font-family: Arial, Helvetica, sans-serif;
    font-weight: 600;
`;

export let Form = styled.form`
  text-align: center;
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  transform: translate(0, -50%);
  `;

export let Main = styled.div`
  height: 90vh;
  position: relative;
`;