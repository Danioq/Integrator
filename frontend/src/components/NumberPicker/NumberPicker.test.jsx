import NumberPicker from './NumberPicker';
import renderer from 'react-test-renderer';
import React from 'react';

describe('NumberPicker component suite', () => {
  test('Render test', () => {
    let component = renderer.create(<NumberPicker getData={jest.fn()} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});