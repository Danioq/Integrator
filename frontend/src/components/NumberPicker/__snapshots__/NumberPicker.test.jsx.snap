// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberPicker component suite Render test 1`] = `
<div
  className="sc-AxgMl hMOoSi"
>
  <form
    className="sc-AxhUy bRctOD"
  >
    <label
      className="sc-AxiKw jLiVrU"
      htmlFor="quantity"
    >
      Wielkość Grupy
    </label>
    <p
      className="sc-AxhCb gQrvhH"
      id="value"
    />
    <input
      className="sc-AxirZ ciOAJL"
      id="quantity"
      max="5"
      min="2"
      name="quantiy"
      onInput={[Function]}
      type="range"
    />
    <input
      className="sc-AxjAm buyOcO"
      onClick={[Function]}
      type="submit"
      value="Wybierz"
    />
  </form>
</div>
`;
